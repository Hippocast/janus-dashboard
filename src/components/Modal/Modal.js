import React from 'react'
import PropTypes from 'prop-types'
import Modaliz from 'react-modaliz'

import block from '../../helpers/bem-cn'

import './Modal.css'

const propTypes = {
  closeModal: PropTypes.func.isRequired,
  message: PropTypes.string.isRequired,
  buttons: PropTypes.arrayOf(PropTypes.node),
  show: PropTypes.bool.isRequired,
  title: PropTypes.string
}

const defaultProps = {
  message: ''
}

const b = block('j-confirmation')

const Modal = ({
  closeModal,
  message,
  buttons,
  show,
  title
}) => (
  <Modaliz
    className={b()}
    show={show}
    speed={500}
    onClose={closeModal}
    discardDefaults
  >
    <div className={b('inner')()}>
      <div className={b('title')()}>
        {title}
      </div>
      <div className={b('body')()}>
        <div className={b('text')()}>
          {message}
        </div>
      </div>
    </div>
    <div className={b('buttons-group').mix('j-buttons__wrapper')()}>
      { buttons }
    </div>
  </Modaliz>
)

Modal.propTypes = propTypes
Modal.defaultProps = defaultProps

export default Modal
