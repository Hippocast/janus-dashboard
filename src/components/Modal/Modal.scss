@import 'variables.scss';

.j-confirmation {
    &-wrapper {
        position: fixed;
        width: 100vw;
        height: 100vh;
        z-index: 1000;
        top: 0px;
        left: 0px;
        opacity: 1;
        transition: 500ms;
    }

    &-container {
        display: flex;
        flex-direction: column;
        padding: $hf-padding-middle $hf-padding-main;
        justify-content: space-between;
        position: absolute;
        max-width: 700px;
        min-height: 100px;
        width: 90vw;
        border-radius: $hf-border-radius;
        box-shadow: $hf-box-shadow;
        background-color: $hf-color-white;
        z-index: 1;
        @include alignCenter();
    }

    &-close {
        position: absolute;
        width: 1.5rem;
        height: 1.5rem;
        top: $hf-padding-main;
        right: $hf-padding-main;
        font-size: 30px;
        display: flex;
        align-items: center;
        justify-content: center;
        cursor: pointer;
        z-index: 1;
    }

    &-overlay {
        position: absolute;
        width: 100%;
        height: 100%;
        background-color: rgba(0, 0, 0, 0.45);
        top: 0px;
        left: 0px;
        z-index: 0;
    }

    &__inner {}
    &__title {
        font-size: $hf-font-size-title;
        line-height: $hf-line-height-title;
        text-align: left;
        color: $hf-color-black;
    }
    &__body {
        padding-top: $hf-padding-big;
        padding-bottom: $hf-padding-big;
    }
    &__text {
        font-size: $hf-font-size-medium;
        line-height: $hf-line-height-medium;
        text-align: left;
        color: $hf-color-gray-medium;
    }

    &__buttons-group {
        justify-content: flex-end;
    }
}
