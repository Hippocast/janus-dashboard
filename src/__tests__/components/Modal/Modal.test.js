/* eslint-env jest */
import React from 'react'
import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json'

import Modal from '../../../components/Modal/Modal'

describe('Modal component', () => {
  const render = props => mount(<Modal {...props} />)
  const mockFunction = jest.fn()
  const requiredProps = {
    closeModal: mockFunction,
    show: true,
    message: 'Modal test message'
  }

  const runCommonTest = (wrapper, props) => {
    const tree = toJSON(wrapper)
    const [modaliz] = tree.children

    expect(tree.props.show).toEqual(props.show)
    expect(
      wrapper.find('.j-confirmation__text').text()
    ).toEqual(props.message)

    modaliz.props.onClose()
    expect(mockFunction).toHaveBeenCalled()
    expect(tree).toMatchSnapshot()
  }

  beforeEach(() => {
    mockFunction.mockReset()
  })

  it('renders correctly', () => {
    const props = requiredProps
    const wrapper = render(props)

    runCommonTest(wrapper, props)
  })

  it('renders correctly with title if an additional prop `title` is passed', () => {
    const props = {
      ...requiredProps,
      title: 'Modal test title'
    }
    const wrapper = render(props)

    runCommonTest(wrapper, props)
    expect(
      wrapper.find('.j-confirmation__title').text()
    ).toEqual(props.title)
  })

  it('renders correctly with buttons if an additional prop `buttons` is passed', () => {
    const props = {
      ...requiredProps,
      buttons: [
        <button key={1}>Cancel</button>,
        <button key={2}>Confirm</button>
      ]
    }
    const wrapper = render(props)

    runCommonTest(wrapper, props)
    expect(
      wrapper.find('.j-confirmation__buttons-group').containsAllMatchingElements(props.buttons)
    ).toBe(true)
  })
})
