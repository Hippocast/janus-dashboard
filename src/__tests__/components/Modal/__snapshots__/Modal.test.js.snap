// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders correctly 1`] = `
<Modal
  closeModal={[Function]}
  message="Modal test message"
  show={true}
>
  <Modaliz
    className="j-confirmation"
    conditionalRender={false}
    discardDefaults={true}
    onClose={[Function]}
    show={true}
    simple={false}
    speed={500}
    text=""
    title=""
  >
    <div
      className="modaliz-wrapper j-confirmation-wrapper"
      style={Object {}}
    >
      <div
        className="modaliz-container j-confirmation-container"
      >
        <div
          className="modaliz-close j-confirmation-close"
          onClick={[Function]}
        >
          ×
        </div>
        <div
          className="j-confirmation__inner"
        >
          <div
            className="j-confirmation__title"
          />
          <div
            className="j-confirmation__body"
          >
            <div
              className="j-confirmation__text"
            >
              Modal test message
            </div>
          </div>
        </div>
        <div
          className="j-confirmation__buttons-group j-buttons__wrapper"
        />
      </div>
      <div
        className="modaliz-overlay j-confirmation-overlay"
        onClick={[Function]}
      />
    </div>
  </Modaliz>
</Modal>
`;

exports[`Modal component renders correctly with buttons if an additional prop \`buttons\` is passed 1`] = `
<Modal
  buttons={
    Array [
      <button>
        Cancel
    </button>,
      <button>
        Confirm
    </button>,
    ]
  }
  closeModal={[Function]}
  message="Modal test message"
  show={true}
>
  <Modaliz
    className="j-confirmation"
    conditionalRender={false}
    discardDefaults={true}
    onClose={[Function]}
    show={true}
    simple={false}
    speed={500}
    text=""
    title=""
  >
    <div
      className="modaliz-wrapper j-confirmation-wrapper"
      style={Object {}}
    >
      <div
        className="modaliz-container j-confirmation-container"
      >
        <div
          className="modaliz-close j-confirmation-close"
          onClick={[Function]}
        >
          ×
        </div>
        <div
          className="j-confirmation__inner"
        >
          <div
            className="j-confirmation__title"
          />
          <div
            className="j-confirmation__body"
          >
            <div
              className="j-confirmation__text"
            >
              Modal test message
            </div>
          </div>
        </div>
        <div
          className="j-confirmation__buttons-group j-buttons__wrapper"
        >
          <button
            key="1"
          >
            Cancel
          </button>
          <button
            key="2"
          >
            Confirm
          </button>
        </div>
      </div>
      <div
        className="modaliz-overlay j-confirmation-overlay"
        onClick={[Function]}
      />
    </div>
  </Modaliz>
</Modal>
`;

exports[`Modal component renders correctly with title if an additional prop \`title\` is passed 1`] = `
<Modal
  closeModal={[Function]}
  message="Modal test message"
  show={true}
  title="Modal test title"
>
  <Modaliz
    className="j-confirmation"
    conditionalRender={false}
    discardDefaults={true}
    onClose={[Function]}
    show={true}
    simple={false}
    speed={500}
    text=""
    title=""
  >
    <div
      className="modaliz-wrapper j-confirmation-wrapper"
      style={Object {}}
    >
      <div
        className="modaliz-container j-confirmation-container"
      >
        <div
          className="modaliz-close j-confirmation-close"
          onClick={[Function]}
        >
          ×
        </div>
        <div
          className="j-confirmation__inner"
        >
          <div
            className="j-confirmation__title"
          >
            Modal test title
          </div>
          <div
            className="j-confirmation__body"
          >
            <div
              className="j-confirmation__text"
            >
              Modal test message
            </div>
          </div>
        </div>
        <div
          className="j-confirmation__buttons-group j-buttons__wrapper"
        />
      </div>
      <div
        className="modaliz-overlay j-confirmation-overlay"
        onClick={[Function]}
      />
    </div>
  </Modaliz>
</Modal>
`;
